### PR TITLE
RowBox fix and BoxExpression refactor...

### PR DESCRIPTION
**Pull request by rocky**
Move BoxExpression out of `mathics.builtin.base` and into mathics.box.expression

BoxExpression didn't have has_form and therefore RowBoxing was giving us too-small boxes when put in a Table.

BoxExpression now has `.elements` just like an Expression does.

Deprecated `BoxConstruct` removed.

### CODE REVIEW

File: mathics/builtin/box/expression.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            options = ListExpression(*elements).get_option_values(evaluation)
+            default.update(options)
+        else:
+            from mathics.core.parser import parse_builtin_rule
+
+            default = {}
+            for option, value in self.options.items():
+                option = ensure_context(option)
+                default[option] = parse_builtin_rule(value)
+        return default
```
Comment (mmatera) [inline mathics/builtin/box/expression.py:203]: Add this line at the end:
```
DOES_NOT_ADD_BUILTIN_DEFINITION = [BoxExpression]
```

Comment (rocky) [reply mathics/builtin/box/expression.py:203]: Done in f3c52a1

File: mathics/format/mathml.py [hunk 2]
```diff
@@ -235,10 +236,10 @@ def is_list_interior(content):
 
     is_list_row = False
     if (
-        len(self.items) == 3
-        and self.items[0].get_string_value() == "{"  # nopep8
+        len(self.items) >= 3
+        and self.items[0].get_string_value() == "{"
         and self.items[2].get_string_value() == "}"
-        and self.items[1].has_form("RowBox", 1)
+        and self.items[1].has_form("RowBox", 1, None)
```
Comment (rocky) [inline mathics/format/mathml.py:242]: Although this PR corrects  `has_form()`  for BoxExpressions, I think all of this code needs to be replaced with something more robust.

**Review by mmatera — APPROVED** [review]
LGTM

**Pull request merged**